core/src/main/scala/com/mesosphere/usi/core/matching/RangeResource.scala

```diff
@@ -0,0 +1,251 @@
+package com.mesosphere.usi.core.matching
+
+import com.mesosphere.usi.core.ResourceUtil
+import com.mesosphere.usi.core.matching.RangeResource._
+import com.mesosphere.usi.core.models.{ResourceMatchResult, ResourceRequirement, ResourceType}
+import com.mesosphere.usi.core.protos.ProtoBuilders
+import org.apache.mesos.v1.Protos
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.util.Random
+
+/**
+  * Represents requirement for Mesos resource of type Range.
+  * You can either request static values (by providing concrete value) or dynamic values (by providing 0).
+  * Dynamic values are then selected from the offered ranges. If random implementation is passed in, these ranges are randomized.
+  * For more information about randomization see [[lazyRandomValuesFromRanges()]]
+  *
+  * @param requestedValues values pod wants to consume on the given ranges
+  * @param resourceType name of resource (e.g. ports)
+  * @param random when requesting dynamic values, you can provide Random implementation if you want dynamic values to be randomized
+  */
+case class RangeResource(requestedValues: Seq[Int], resourceType: ResourceType, random: Random = Random)
+    extends ResourceRequirement {
+  override def description: String = s"$resourceType:[${requestedValues.mkString(",")}]"
+
+  override def matchAndConsume(resources: Seq[Protos.Resource]): Option[ResourceMatchResult] = {
+    matchAndConsumeIter(Nil, resources.toList)
+  }
+
+  /**
+    * Iterates over given resources keeping in track unmatched resources.
+    * In the end produces ResourceMatchResult if we were able to find a match on the given resources.
+    *
+    * @param unmatchedResources already processed resources that did not contain resources we are trying to match
+    * @param remainingResources resources we still need to process when looking for resource match
+    * @return None if no match was possible, ResourceMatchResult if we were able to consume requested resources
+    */
+  @tailrec private def matchAndConsumeIter(
```

Comment:
super nit: the same method in `ScalarResource` is called just `iter` ;)

Reply:
yeah I know ... I like it more like this :slightly_smiling_face: I think it's easier to understand...